packages/eslint-plugin/src/rules/jsx-pragma/index.ts

```diff
@@ -144,14 +154,20 @@ export const jsxPragmaRule: Rule.RuleModule = {
     const optionsRaw = context.options[0] || {};
     const options: Options = {
       detectConflictWithOtherLibraries: optionsRaw.detectConflictWithOtherLibraries ?? true,
-      onlyRunIfImportingCompiled: optionsRaw.onlyRunIfImportingCompiled ?? false,
+      onlyRunIfImportingCompiled:
+        optionsRaw.onlyRunIfImportingCompiled ?? optionsRaw.importSources ?? false,
```

Comment:
Should this at all be checking length, do you get into a weird place where `onlyRunIfImportingCompiled = […]` and it should be a boolean?  Probably doesn't matter, just I think this would fail type-checking?

eg.
```suggestion
        optionsRaw.onlyRunIfImportingCompiled ?? !!optionsRaw.importSources?.length || false,
```

Reply:
ah that's a good point, `optionsRaw.importSources` was typed as `any` (it's an eslint thing 😅 ) so it didn't get caught

it doesn't affect the execution of the rule to my knowledge (except for the edge case where `importSources = []`), but your suggestion does reduce confusion